### ISSUE TITLE
Memcached and Selection popups

### ISSUE THREAD

**Issue by markbond1007** [salesagility/SuiteCRM#1248]
I have Memcached installed on my server (and running), along with the pear php module for it. With this all installed when I Click on any of the "Select" options to select items to relate the pop-up page fails to load and errors get dumped into the apache log:

`apache2: libmemcached/storage.cc:341: memcached_return_t memcached_send_ascii(memcached_st*, memcached_server_write_instance_st, const char*, size_t, const char*, size_t, time_t, uint32_t, uint64_t, bool, bool, memcached_storage_action_t): Assertion `memcached_failed(rc)' failed.
apache2: libmemcached/storage.cc:341: memcached_return_t memcached_send_ascii(memcached_st*, memcached_server_write_instance_st, const char*, size_t, const char*, size_t, time_t, uint32_t, uint64_t, bool, bool, memcached_storage_action_t): Assertion `memcached_failed(rc)' failed.
apache2: libmemcached/storage.cc:341: memcached_return_t memcached_send_ascii(memcached_st*, memcached_server_write_instance_st, const char*, size_t, const char*, size_t, time_t, uint32_t, uint64_t, bool, bool, memcached_storage_action_t): Assertion `memcached_failed(rc)' failed.
apache2: libmemcached/storage.cc:341: memcached_return_t memcached_send_ascii(memcached_st*, memcached_server_write_instance_st, const char*, size_t, const char*, size_t, time_t, uint32_t, uint64_t, bool, bool, memcached_storage_action_t): Assertion `memcached_failed(rc)' failed.`

So for example go to target lists, edit a specific list, scroll down to contacts select the options labeled 'Select' the pop up window appears with an Empty Response error and the above errors getting logged into apache.

Memcached appears to be working in other locations okay!

**Comment by vladbar** [comment]
Hello markbond1007, have you tried a cache flush.

**Comment by markbond1007** [comment]
Yes I tried that, the only thing to fix is to disable it completely. Memcached itself works and test code accessing it (via PHP) works.

**Comment by terencelhl** [comment]
Thank you for the solution. Am having the same problem too.

**Comment by dsmikhal** [comment]
Hi Mark,

can you please advice me - how did you disable memcached? did you stop a service at server level or disabled it for this exact CRM instance? I have the same issue, but there are multiple CRM runs on the server and I need to disable it only for one instance...

**Comment by chris001** [comment]
It's incredible how a bug as important to the smooth running of the application as this has gone for a year, with no comment from @salesagility , and nobody assigned to fix it.
@dsmikhal To disable `memcached`, you must go into `config_override.php` and add:
`$sugar_config['external_cache_disabled_memcached'] = true;`

**Comment by dsmikhal** [comment]
Thanks Chris, it works!

**Comment by chris001** [comment]
Good, Dmitrii, keep in mind, disabling `memcached` made your users experience in the application suddenly become painfully slow.  This `memcached` bug is still unfixed.  Once @salesagility will merge #3007 and #3237, it should be fixed sooner.  If anyone is reading this, the key to adapt the ancient, outdated `SugarCache` classes to use PSR-6 from issue #3030.
